DemoEdgeApp/ModuleWrapper/TaskTimer.cs

```diff
@@ -0,0 +1,72 @@
+namespace ModuleWrapper
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Serilog;
+
+    public class TaskTimer
+    {
+        CancellationTokenSource cancellationTokenSource;
+        TimeSpan timerPeriod;
+        Action onElapsedCallback;
+        bool continueOnError;
+
+        public TaskTimer(Action onElapsedCallback,
+            TimeSpan timerPeriod,
+            CancellationTokenSource cancellationTokenSource,
+            bool continueOnError = true)
+        {
+            this.cancellationTokenSource = cancellationTokenSource;
+            this.timerPeriod = timerPeriod;
+            this.onElapsedCallback = onElapsedCallback;
+            this.continueOnError = continueOnError;
+        }
+
+        public void Start()
+        {
+            Task elapsedTask = null;
+            elapsedTask = new Task((x) =>
+            {
+                Elapsed(elapsedTask, cancellationTokenSource);
+            }, cancellationTokenSource.Token);
+
+            HandleError(elapsedTask);
+
+            elapsedTask.Start();
+        }
+
+        private void Elapsed(Task task, object objParam)
+        {
+            var start = DateTime.Now;
+            var cancellationTokenSource = (CancellationTokenSource)objParam;
+            if (cancellationTokenSource.Token.IsCancellationRequested)
+            {
+                Log.Information("A cancellation has been requested.");
+                return;
+            }
+
+            onElapsedCallback();
+
+            var delay = timerPeriod - (DateTime.Now - start);
```

Comment:
two options for private member field names: prefix with an _ to signify it's a private variable (as opposed to a local) or always add this to the front to clarify that you are referencing a local variable. So timerPeriod here would be this.timerPeriod or _timerPeriod

Reply:
agree, fixed